https://bugs.gentoo.org/921720
https://invent.kde.org/utilities/kate/-/issues/116
https://invent.kde.org/utilities/kate/-/merge_requests/1387
https://invent.kde.org/utilities/kate/-/merge_requests/1388

From 83fcf7b4e7cc7bc54828192e9b89cfb286d864fd Mon Sep 17 00:00:00 2001
From: Aleix Pol <aleixpol@kde.org>
Date: Tue, 19 Sep 2023 16:34:49 +0200
Subject: [PATCH] Fix build

Ever since I last updated, I've started to get errors about includes
missing, so I've added them.

(cherry picked from commit 73d090f7f7cb7cc013f31c7b572ca78a7fbeb5b8)
--- a/addons/git-blame/kategitblameplugin.h
+++ b/addons/git-blame/kategitblameplugin.h
@@ -20,6 +20,7 @@
 #include <QLocale>
 #include <QPointer>
 #include <QTimer>
+#include <QUrl>
 #include <QVariant>
 #include <QVector>
 
--- a/addons/project/git/gitstatus.h
+++ b/addons/project/git/gitstatus.h
@@ -5,6 +5,7 @@
 */
 #pragma once
 
+#include <QSet>
 #include <QString>
 #include <QVector>
 
--- a/apps/lib/ktexteditor_utils.h
+++ b/apps/lib/ktexteditor_utils.h
@@ -7,6 +7,8 @@
 #include "kateprivate_export.h"
 
 #include <QString>
+#include <QUrl>
+#include <QWidgetList>
 
 QT_BEGIN_NAMESPACE
 class QScrollBar;
-- 
GitLab

From bd8403ba1771d92586518868d0bfd42087b0db4d Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Fri, 12 Jan 2024 16:53:56 +0000
Subject: [PATCH] Add further missing includes

Bug: https://bugs.gentoo.org/921935
Bug: https://invent.kde.org/utilities/kate/-/issues/116
Signed-off-by: Sam James <sam@gentoo.org>
--- a/addons/project/gitstatusmodel.h
+++ b/addons/project/gitstatusmodel.h
@@ -5,6 +5,7 @@
 */
 #pragma once
 
+#include <QSet>
 #include <QAbstractItemModel>
 
 #include "git/gitstatus.h"
--- a/addons/project/kateproject.h
+++ b/addons/project/kateproject.h
@@ -7,6 +7,8 @@
 
 #pragma once
 
+#include <QtGlobal>
+
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
 #include <KTextEditor/ModificationInterface>
 #endif
--- a/addons/project/kateprojectitem.h
+++ b/addons/project/kateprojectitem.h
@@ -7,6 +7,8 @@
 
 #pragma once
 
+#include <QtGlobal>
+
 #if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
 #include <KTextEditor/Document>
 #else
-- 
GitLab
